### PR TITLE
[7.4 only] LPS-122456 - Remove {append, buildFragment, enterDocument} usages of metal-dom

### PR DESCRIPTION
**Pull request by bryceosterhaus**
follow up from https://github.com/liferay-frontend/liferay-portal/pull/463

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
@john-co I also can't seem to figure out why this one is breaking as well... I looked into the poshi report, but locally it seems fine again, so I'm not sure if this is a flakey test or something else.

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
There seems to have been an unrelated issue prior to running some functional tests...can you fix the merge conflict and run the tests again?

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by john-co** [comment]
All the functional tests are failing in the latest test run. Seems like dropdown menus are not visible after clicking on some ellipsis.

image

**Comment by jbalsas** [comment]
Makes sense this will fail since [createContextualFragment is Chrome 86+](https://caniuse.com/?search=createContextualFragment) and we're running Chrome 65 on CI 😢

**Comment by wincent** [comment]
> Makes sense this will fail since [createContextualFragment is Chrome 86+](https://caniuse.com/?search=createContextualFragment) and we're running Chrome 65 on CI 😢

We've been using `createContextualFragment` in a number of places now for at least a year (I'm too lazy to look in the Git log to see exactly when the first usage entered the repo), so that either means:

- Our test coverage wasn't hitting any of those code paths; or:
- Chrome 65's support is there but patchy in a way that the problem only manifests now with these new usages; or:
- There is something wrong with our implementation here.

(Note that caniuse just labels versions 4 to 85 as "support unknown"; there's _no way_ that support only just arrived in v86, which came out a month ago).

**Comment by jbalsas** [comment]
> We've been using createContextualFragment in a number of places now for at least a year (I'm too lazy to look in the Git log to see exactly when the first usage entered the repo)

Yeah, this also striked as odd to me. I think the reason is likely that we weren't hitting those code paths in our tests and now we're expanding the possibilities.

**Comment by jbalsas** [comment]
I've rebased to fix the merge conflicts and force pushed... let's give this another try 😢

**Comment by jbalsas** [comment]
ci:test:relevant

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js [hunk 1]
```diff
@@ -70,8 +70,7 @@ class DynamicInlineScroll extends PortletBase {
 	addListItem_(listElement, pageIndex) {
 		const listItem = document.createElement('li');
 
-		dom.append(
-			listItem,
+		listItem.append(
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js:73]: I don't think these replacements are correct:

> The ParentNode.append() method inserts a set of Node objects or DOMString objects after the last child of the ParentNode. DOMString objects are inserted as equivalent Text nodes.

`DOMString` objects are inserted as such and not transformed to `Node` objects. Probably need to build the fragments and maybe we can always use `appendChild` so there's no confusion on what the API does and takes.

Maybe this is the reason for the test failures we're seeing consistently in this PR.

/cc @bryceosterhaus, @diegonvs

Comment (bryceosterhaus) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js:73]: @jbalsas good catch. I'll go through and update.

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
Added mocks for `createRange` and `createContextualFragment`. Ideally these get added globally [here](https://github.com/liferay/liferay-frontend-projects/pull/234), but want to get this tested and merged as soon as possible, so we can always remove these mocks later in favor of global mock via npm scripts

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:stop

**Comment by bryceosterhaus** [comment]
This PR is gonna be dependent on the jest mocks provided with the new npm-scripts at https://github.com/liferay-frontend/liferay-portal/pull/506

**Comment by jbalsas** [comment]
I'm going to close this and rebase on top of https://github.com/liferay-frontend/liferay-portal/pull/529 to remove the Senna changes and maybe even remove them completely as to try to get this unblocked!

**Comment by jbalsas** [comment]
See https://github.com/liferay-frontend/liferay-portal/pull/530

**Pull request closed without merging**